src/spiceypy/spiceypy.py

```diff
@@ -14917,6 +14917,87 @@ def szpool(name: str) -> Tuple[int, bool]:
 ################################################################################
 # T
 
+@spice_error_check
+def tangpt(
+    method: str,
+    target: str,
+    et: float,
+    fixref: str,
+    abcorr: str,
+    corloc: str,
+    obsrvr: str,
+    dref: str,
+    dvec: Union[ndarray, Iterable[float]],
+) -> Tuple[ndarray, float, float, ndarray, float, ndarray]:
+    """
+    Compute, for a given observer, ray emanating from the observer,
+    and target, the "tangent point": the point on the ray nearest
+    to the target's surface. Also compute the point on the target's
+    surface nearest to the tangent point.
+
+    The locations of both points are optionally corrected for light
+    time and stellar aberration.
+
+    The surface shape is modeled as a triaxial ellipsoid.
+
+    https://naif.jpl.nasa.gov/pub/naif/toolkit_docs/C/cspice/tangpt_c.html
+
+    :param method: Computation method.
+    :param target: Name of target body.
+    :param et: Epoch in ephemeris seconds past J2000 TDB.
+    :param fixref: Body-fixed, body-centered target body frame.
+    :param abcorr: Aberration correction.
+    :param corloc: Aberration correction locus: "TANGENT POINT" or
+                   "SURFACE POINT".
+    :param obsrvr: Name of observing body.
+    :param dref: Reference frame of ray direction vector.
+    :param dvec: Ray direction vector.
+    :return: "Tangent point": point on ray nearest to surface, Altitude of
+     tangent point above surface, Distance of tangent point from observer,
+     Point on surface nearest to tangent point, Epoch associated with
+     correction locus, Vector from observer to surface point `srfpt'.
+    """
+    method = stypes.string_to_char_p(method)
```

Comment:
okay so I haven't reformatted the rest of this file yet but I've developed a new convention where all parameter type conversion should assign to variables prepended by an underscore ie `_method = stypes.string_to_char_p(method)` to help better support mypy type checking (which wants static typing for everything). Repeat for the rest of the parameters below this line